scipy/stats/tests/test_stats.py

```diff
@@ -2735,64 +2735,103 @@ class TestMannWhitneyU(TestCase):
     def test_mannwhitneyu_less(self):
         u1, p1 = stats.mannwhitneyu(self.X, self.Y, alternative='less')
         u2, p2 = stats.mannwhitneyu(self.Y, self.X, alternative='greater')
+        u3, p3 = stats.mannwhitneyu(self.Y, self.X, alternative='less')
 
         assert_equal(p1, p2)
         assert_equal(u1, 498)
-        assert_equal(u2, 102)
+        assert_equal(u2, 498)
+        assert_equal(u3, 102)
         assert_approx_equal(p1, 0.999957683256589, significant=self.significant)
 
     def test_mannwhitneyu_greater(self):
         u1, p1 = stats.mannwhitneyu(self.X, self.Y, alternative='greater')
         u2, p2 = stats.mannwhitneyu(self.Y, self.X, alternative='less')
+        u3, p3 = stats.mannwhitneyu(self.Y, self.X, alternative='greater')
 
         assert_equal(p1, p2)
-        assert_equal(u1, 498)
+        assert_equal(u1, 102)
         assert_equal(u2, 102)
+        assert_equal(u3, 498)
         assert_approx_equal(p1, 4.5941632666275e-05,
                             significant=self.significant)
 
     def test_mannwhitneyu_two_sided(self):
         u1, p1 = stats.mannwhitneyu(self.X, self.Y, alternative='two-sided')
-        u2, p2 = stats.mannwhitneyu(self.Y, self.X)  # two-sided is default
+        u2, p2 = stats.mannwhitneyu(self.Y, self.X, alternative='two-sided')
 
         assert_equal(p1, p2)
-        assert_equal(u1, 498)
+        assert_equal(u1, 102)
         assert_equal(u2, 102)
         assert_approx_equal(p1, 9.188326533255e-05,
                             significant=self.significant)
 
+    def test_mannwhitneyu_default(self):
+        # The default value for alternative is None
+        u1, p1 = stats.mannwhitneyu(self.X, self.Y)
+        u2, p2 = stats.mannwhitneyu(self.Y, self.X)
+        u3, p3 = stats.mannwhitneyu(self.X, self.Y, alternative=None)
+
+        assert_equal(p1, p2)
+        assert_equal(p1, p3)
+        assert_equal(u1, 102)
+        assert_equal(u2, 102)
+        assert_equal(u3, 102)
+        assert_approx_equal(p1, 4.5941632666275e-05,
+                            significant=self.significant)
+
     def test_mannwhitneyu_no_correct_less(self):
         u1, p1 = stats.mannwhitneyu(self.X, self.Y, False, alternative='less')
         u2, p2 = stats.mannwhitneyu(self.Y, self.X, False,
                                     alternative='greater')
+        u3, p3 = stats.mannwhitneyu(self.Y, self.X, False,
+                                    alternative='less')
 
         assert_equal(p1, p2)
         assert_equal(u1, 498)
-        assert_equal(u2, 102)
+        assert_equal(u2, 498)
+        assert_equal(u3, 102)
         assert_approx_equal(p1, 0.999955905990004, significant=self.significant)
 
     def test_mannwhitneyu_no_correct_greater(self):
```

Comment:
It looks to me this test and the previous one could be merged because they overlap

(streamlining, minor point)